### ISSUE TITLE
More GUI Updates

### ISSUE THREAD

**Issue by QartemisT** [DeadlyBossMods/DBM-Unified#3]
- [ ] Splitting content mods into two different tabs (raid & pvp/dungeon & scenario)
- [ ] Add a separte tab support for plugins that fit outside content mods
- [ ] Add "About" tab
- [ ] GUI Alpha (Transparency) options
- [ ] InfoFrame functions rewrite
- [ ] Global settings wipe (Factory reset, with a confirm option... Re-init from table copy, and wipe settings table)
- [x] Import/Export feature (ref own issue https://github.com/DeadlyBossMods/DeadlyBossMods/issues/321)
- [x] Timer Appearance panel may include skins stuff there or it may be a 3rd panel for skins instead as

----
Myst requests:
- [ ] Add input box to color pickers that allows users to copy/enter color hex code.
- [ ] Add infoframe role icons
- [ ] Add infoframe separators for infoframe types that are numeric in nature (such as stack and table)
Example:
all 6 stacks
-line-
all 5 stacks
-line-
etc

**Comment by linaori** [comment]
### Additional ideas

#### Shortcut to the encounter from the journal

The red squares could have a button to open the encounter directly in the DBM settings
![image](https://github.com/DeadlyBossMods/DBM-Unified/assets/1754678/ae7daf5b-c1bf-4947-9ce4-7a6ffbd8b0a3)

#### Shortcut to the trash or boss encounter in MDT

Plater has recently started adding shortcuts to NPC health colors and cast colors in MDT. DBM could do something similar for the NPC/Casts if there is something for that NPC or cast and have it open the DBM settings directly.

![image](https://github.com/DeadlyBossMods/DBM-Unified/assets/1754678/64088479-8e58-49a6-be74-2fdcc58238c2)

#### Mini settings frame
On top of the above, it could make it much easier to configure a specific ability by having a button (or click or whatever) from MDT to open a mini settings frame for that specific mechanic or cast in addition to having a way to open the settings in DBM (previous suggestion)

![image](https://github.com/DeadlyBossMods/DBM-Unified/assets/1754678/3404e01a-a9eb-4d5d-980b-2b501a0910a2)

**Comment by MysticalOS** [comment]
Post merge of existing PR, need to think about how to better approach other category, as it's sort of a dumping around for "everything else"
There is a lot of everything else though, off top of head is
World Events
PVP
Mist of Pandaria Scenarios Mod
Challenges
Brawlers (will probably move into challenges, or did I do that already?)
Delves in 11.x

I guess that's not too many to have there for now. none of those fit worthyness of needing a whole new tab, at least right now, but if delves remain evergreen then in a later expansion, when there are multiple I can see Delves (and mop scenarios cause they are similar concept) getting own content tab in like 12.0 or 13.0

**Comment by linaori** [comment]
@MysticalOS I agree that it should get more categories afterwards as it will make navigation much easier. There can still be an "other" tab for things that really don't have a category, think of the logout timer for example, but I'm also not opposed to adding  a "Plugin Options" (or "Mod Options") next to "Core Options", or adding a "Plugin Options" inside the Core options tab itself as a new header.

**Comment by MysticalOS** [comment]
I thought about a plugins tab at one point, but issue now is there just aren't enough to merit a whole tab. raid lead tools is no longer maintained and quite frankly woefully inadequate even if it was. spelltimers also has super low adoption cause again better options exist, and archaeology is more of an install and forget it. It's basically a tab for something that just doesn't really exist. So i think only world bosses needs a tab for now (and that's on todo for existing PR) and then in future, probably delves/scenario content.
everything else can sit in other for now (and spelltimers can just inject into core options like it's always done)

**Comment by MysticalOS** [comment]
![Screenshot 2024-01-07 at 11 57 15 PM](https://github.com/DeadlyBossMods/DBM-Unified/assets/1899149/7bdc69b5-2fe8-4062-aa52-2a6044b4be1c)
$spell not rendered in some cases. not sure if classic only bug or just bug with this specific object.

**Comment by MysticalOS** [comment]
DBM GUI should auto expand expand the raid or dungeon category when doing /dbm if you are inside

IE a code that basically goes
OnShow, if lastInstanceID == a loaded module then
if not expanded, expand end

it should just be assumed that if a player is doing /dbm in a raid or in a dungeon, they probably are going in there to check/change something for that dungeon, so lets make it easier with less clicks

**Comment by MysticalOS** [comment]
When opening GUI for first time, instead of selecting no panel by default, it should select general options header as default

IE look like this
![Screenshot 2024-01-11 at 7 19 06 PM](https://github.com/DeadlyBossMods/DBM-Unified/assets/1899149/442b755c-313f-41f6-9802-68222b4eff38)
This way, the first thing players see are the initial setup wiki guides. Obviously after that it should remember last visible. it should only be set to this panel on first show.
